### DISCUSSION TITLE
Audiovisual MediaStreams are destroyed in some networks: "NOVIABLEICE"

### DISCUSSION THREAD

**Discussion by jimver04** [networked-aframe/networked-aframe#412] [Q&A] [answered]
Hi Vincent, 

in general NAF is working well in any network I have tried. However,

@Mobile, Android 11, Mobile Chrome or Firefox, in a certain public network I get "NOVIABLEICE" and easy-rtc closes MediaStreams. I see that entities are created across peers but mediastreams are automatically destroyed. Any idea ?

This does not happen using Chrome@PC  in this network. I suspect that 

- there is a setting in mobile devices for such networks and for MediaStreams specifically ?

I will look into it. 

**Related resources:**
1. https://stackoverflow.com/questions/49470556/how-to-fix-noviableice-error-in-easyrtc
2. https://github.com/networked-aframe/networked-aframe/issues/10
3. https://groups.google.com/g/easyrtc/c/iYV4QLJ96ks

Also @hthetiot  might know the answer.

My easyrtc-server.js contains the following setting regarding ice/stun

```
// Start Socket.io so it attaches itself to Express server
const socketServer = socketIo.listen(webServer, {"log level": 1});
const myIceServers = [
  {"urls":"stun:stun1.l.google.com:19302"},
  {"urls":"stun:stun2.l.google.com:19302"},
  // {
  //   "urls":"turn:[ADDRESS]:[PORT]",
  //   "username":"[USERNAME]",
  //   "credential":"[CREDENTIAL]"
  // },
  // {
  //   "urls":"turn:[ADDRESS]:[PORT][?transport=tcp]",
  //   "username":"[USERNAME]",
  //   "credential":"[CREDENTIAL]"
  // }
];
```

Best,
Dimitrios

**Comment by vincentfretin** [comment, marked as answer]
Hi,
The message is self explanatory, there is no viable ice candidate to make the connection.
You need to configure a turn server, see
https://github.com/networked-aframe/networked-aframe/blob/master/docs/hosting-networked-aframe-on-a-server.md#extra-steps-for-webrtc

**Comment by jimver04** [comment]
I have found a public turn server that did the job. It is useful for those that want to run NAF in public wifi networks using mobile devices !!!


:smirk:

```
const myIceServers = [
    {
        urls: "turn:openrelay.metered.ca:80",
        username: "openrelayproject",
        credential: "openrelayproject",
    }
```
